[kngys] iP

kngys opened the pull request:
# Serenity

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Serenity frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn 
- ~~FAST~~ _SUPER_ FAST to use 

All you need to do is,

1. download it from [here](https://github.com/kngys/ip.git)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 💯 

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:


```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kngys committed:
Level 0. Rename, Greet, Exit

kngys committed:
Level 1. Echo

kngys committed:
Update to accept user input as a phrase instead of only a single word

kngys committed:
Level 2. Add, List

kngys committed:
Add Task class

kngys committed:
Level 3. Mark as Done

kngys committed:
Add Deadline class

kngys committed:
Add Todo class

kngys committed:
Add Event class

kngys committed:
Level 4. ToDos, Events, Deadlines

kngys committed:
A-TextUiTesting

kngys committed:
Level 5. Handle Errors

kngys committed:
Level 6. Delete

kngys committed:
Add method to format saving of tasks

kngys committed:
Add feature to save tasks

kngys committed:
Merge branch 'branch-Level-7'

kngys committed:
Store deadline dates as LocalDate and format date

kngys committed:
Merge branch 'branch-Level-8'

kngys committed:
Add Ui, Storage, Parser and TaskList classes

kngys committed:
Update logic of main class

kngys committed:
Merge branch 'branch-A-MoreOOP'

kngys committed:
Put all classes in one package

kngys committed:
Merge branch 'branch-A-Packages'

kngys committed:
Merge branch 'add-gradle-support'

kngys committed:
Add Gradle

kngys committed:
Add JUnit tests to test Todo, TaskList and Storage classes

kngys committed:
Merge branch 'branch-A-JUnit'

kngys committed:
Package app as JAR file

kngys committed:
Add JavaDoc comments

kngys committed:
Update to follow coding standard

kngys committed:
Add find function

kngys committed:
Merge branch 'branch-A-JavaDoc'

kngys committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/serenity/Event.java
#	src/main/java/serenity/Parser.java
#	src/main/java/serenity/Storage.java

kngys committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/serenity/TaskList.java

PeterHW963 commented on src/main/java/serenity/TaskList.java:
```diff
@@ -0,0 +1,195 @@
+        Task t;
+
+        if (input.startsWith("todo")) {
+            String[] description = input.split(" ");
```
This naming seems a bit vague. What this line is doing is creating a temporary placeholder array to check if there is anything after the command words "todo", "deadline", "event", and then the `description` array isn't used anymore, am I correct? So perhaps other names like `splitInput `or `preprocessedInput `would make the utility of the variable clearer.

PeterHW963 commented on src/main/java/serenity/TaskList.java:
```diff
@@ -0,0 +1,195 @@
+            throw new SerenityException("Error: Missing task index.");
+        }
+
+        int index = Integer.parseInt(input.substring(7)) - 1;
```
Just a thought, isn't this going to throw unnecessary exceptions when the user double spaces (either accidentally or intentionally)? eg. Input = `delete  2`

PeterHW963 reviewed (commented):
(I'm here for the second PR review - the code naming quality one, sorry but abit laze to comment on other coding standards 😢 not that I can spot much)
Solid naming throughout though 👍

kngys committed:
Update JavaDoc comments

kngys committed:
Add checkstyle configurations

kngys committed:
Update to follow coding standard

kngys committed:
Merge branch 'branch-A-CheckStyle'

Justin-Aw46 commented on src/main/java/serenity/Parser.java:
```diff
@@ -0,0 +1,72 @@
+     * @return True if the command is bye.
+     */
+
+    public static boolean isExit(String input) {
```
This is pretty smart I couldn't think of a way to handle the exit for my iP

Justin-Aw46 commented on src/main/java/serenity/Event.java:
```diff
@@ -0,0 +1,43 @@
+public class Event extends Task {
+
+    protected String start;
+    protected String end;
```
Maybe can be a bit more precise like startTime and endTime

Justin-Aw46 reviewed (commented):
LGTM I couldn't find much coding standard issues :)

Justin-Aw46 commented:
Im just commenting here to make sure I got all the bases covered but good job with the iP bro

kngys committed:
Add GUI functionality

kngys committed:
Merge branch 'branch-Level-10'

kngys committed:
Add assertions

There are currently no assertion statements in the code.

Assertion statements define our assumptions of the program state. These
assumptions are verified at runtime, allowing errors to be caught early.

Let's add some assertion statements to ensure our assumptions about the
program state are true.

kngys committed:
Improve code quality

The current code has areas for improvement in terms of code quality.

Improving code quality allows for code to be better read, understood
and modified by other developers.

Let's
*Shorten methods
*Improve logical structure of code
*Remove unnecessary comments
*Refactor by extracting methods

kngys committed:
Merge pull request #1 from kngys/branch-A-Assertions

Add assertions

kngys committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/serenity/Parser.java

kngys committed:
Merge pull request #2 from kngys/branch-A-CodeQuality

Improve code quality

kngys committed:
Add functionality to edit task details

kngys committed:
Merge branch 'branch-C-Update'

kngys committed:
Update GUI

kngys committed:
Add user guide

kngys committed:
Update user guide

kngys committed:
Update user guide

kngys committed:
Update JavaDoc comments and exception handling

kngys committed:
Update GUI and JUnit tests

kngys committed:
Update user guide

kngys committed:
Improve exception handling

kngys committed:
Improve code quality

kngys committed:
Update user guide

kngys committed:
Improve exception handling

kngys committed:
Update user guide

kngys committed:
Improve code quality
